### ISSUE TITLE
Moq1002 false positive when using expression constructor

### ISSUE THREAD

**Issue by MattKotsenas** [rjmurillo/moq.analyzers#234]
This code fires [Moq1002](https://github.com/rjmurillo/moq.analyzers/blob/main/docs/rules/Moq1002.md) when I believe that it shouldn't

```csharp
_ = new Mock<Calculator>(() => new Calculator(), MockBehavior.Loose);

public class Calculator
{
    public int Add(int a, int b) => a + b;
}
```

Here's the overload that should be used: https://github.com/devlooped/moq/blob/18dc7410ad4f993ce0edd809c5dfcaa3199f13ff/src/Moq/Mock%601.cs#L200
